pkg/db/db.go

```diff
@@ -0,0 +1,117 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+)
+
+func getConnectionString(db *config.DB) string {
+	return fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", db.Host, db.Port, db.User, db.Password, db.Name)
+}
+
+type Database struct {
+	tableName string
+	pool      *pgxpool.Pool
+}
+
+func NewClient(ctx context.Context, db *config.DB) (*Database, error) {
+	connString := getConnectionString(db)
+	pool, err := pgxpool.New(ctx, connString)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := pool.Ping(ctx); err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		pool:      pool,
+		tableName: db.ProvidersTableName,
+	}, nil
+}
+
+func (pg *Database) Close() {
+	pg.pool.Close()
+}
+
+var once sync.Once
```

Comment:
Do we really need an alias for a single use?

Reply:
This is removed now.